### DISCUSSION TITLE
Adding an item to a deeply nested array using $ADD

### DISCUSSION THREAD

**Discussion by testflyjets** [dynamoose/dynamoose#1425] [Q&A] [answered]
I'm trying to append a new item to an array nested inside of another array. In the schema below, I'm trying to add a new item to the `completedLessons` array:

```JSON
{
    "id": "first-course",
    "modules": [
        {
            "module": "learn-stuff",
            "lessonCount": 10,
            "completedLessons": [
                {
                    "name": "lesson-1",
                    "completedDate": 1655956900000
                },
                {
                    "name": "lesson-2",
                    "completedDate": 1655956900000
                }
            ]
        },
        {
             "module": "learn-more-stuff",
             ...
        }
    ]
}
```

My hope was that I could provide the document path to the nested array like this:

`modules[0].completedLessons`

and then use something like this:

```JavaScript
await Course.update({"id": "first-course}, 
    { "$ADD": {
        "modules[0].completedLessons": { 
            name: "lesson-3", 
            completedDate: 1655956999000 }
        }
    }
);
```

I resorted to adding the completed lesson to the array within the entirety of the `modules` array and then updating the `modules` attribute, but it feels like there should be a more efficient way to simply append the new completed lesson to the array.

**Comment by andrewda** [comment, marked as answer]
This is currently a limitation in dynamoose. We had a discussion about this here https://github.com/dynamoose/dynamoose/issues/665#issuecomment-791203352, but I haven't had the time to finish figuring out the implementation. For now I suggest using `aws-sdk` directly, and doing something like:

```
SET modules[0].completedLessons = list_append(modules[0].completedLessons, :value)
```

I've still needed this occasionally as well, so I may try to find the time to finally take a stab at it in the next few weeks. When I do make progress, I'll be sure to update both on that issue as well as here!

**Reply by testflyjets** [reply]
Thanks very much for your quick reply and the suggested workaround. I look forward to seeing this as a feature in Dynamoose.

**Reply by testflyjets** [reply]
@andrewda I've not called the AWS DynamoDB SDK directly before -- is there a way to do that via Dynamoose? Apologies for the newbie question, but if you have a more complete example of how to make that `SET` call with the aws-sdk, that would be awesome.
